lib/posix-process/exit.c

```diff
@@ -0,0 +1,318 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2024, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+#include <errno.h>
+#include <sys/types.h>
+#include <stddef.h>
+
+#include <uk/config.h>
+#include <uk/event.h>
+#include <uk/process.h>
+#include <uk/syscall.h>
+
+#if CONFIG_LIBPOSIX_PROCESS_SIGNAL
+#include "signal/signal.h"
+#include "signal.h"
+#endif /* CONFIG_LIBPOSIX_PROCESS_SIGNAL */
+
+#include "process.h"
+
+#define PTHREAD_WAITING_FOR_PID(_pt, _pid)				\
+	((_pt)->state == POSIX_THREAD_BLOCKED_WAIT &&			\
+	 ((_pt)->wait_pid == (_pid) || (_pt)->wait_pid == UK_PID_WAIT_ANY))
+
+#if CONFIG_LIBPOSIX_PROCESS_SIGNAL
+#define PTHREAD_BLOCKING_ON_SIGNAL(_pt)					\
+	((_pt)->state == POSIX_THREAD_BLOCKED_SIGNAL)
+#else /* !CONFIG_LIBPOSIX_PROCESS_SIGNAL */
+#define PTHREAD_BLOCKING_ON_SIGNAL(_pt)		    0
+#endif /* !CONFIG_LIBPOSIX_PROCESS_SIGNAL */
+
+UK_EVENT(POSIX_PROCESS_EXIT_EVENT);
+
+int pprocess_exit_status;
```

Comment:
I see this is not `static` and is also exported in the header. Is there anyone else using it?

Reply:
Yes, it is used in the upcoming PR for shutdown.